tests/test_serialize_entities.py

```diff
@@ -326,14 +326,39 @@ def __init__(self, message):
             }
         ]
     },
-    "trace_id": segment.trace_id,
+    "trace_id": segment_one.trace_id,
     "fault": True,
-    "end_time": segment.end_time
+    "end_time": segment_one.end_time
     }
 
-    actual_segment_dict = entity_to_dict(segment) 
+    segment_two = Segment('test')
+    subsegment = Subsegment('test', 'local', segment_two)
+
+    subsegment.add_exception(exception_one, stack_one, True)
+    subsegment.add_exception(exception_two, stack_two, False)
+    subsegment.close()
     
-    assert expected_segment_dict == actual_segment_dict
+    # will record cause id instead as same exception already recorded in its subsegment
+    segment_two.add_exception(exception_one, stack_one, True)
+    
+    segment_two.close()
+    
+    expected_segment_two_dict = {
+    "id": segment_two.id,
+    "name": "test",
+    "start_time": segment_two.start_time,
+    "in_progress": False,
+    "cause": exception_one._cause_id,
+    "trace_id": segment_two.trace_id,
+    "fault": True,
+    "end_time": segment_two.end_time
+    }
+
+    actual_segment_one_dict = entity_to_dict(segment_one)
+    actual_segment_two_dict = entity_to_dict(segment_two)
+    
+    assert expected_segment_one_dict == actual_segment_one_dict
```

Comment:
Not a big deal, but we should have two separate test cases since we are testing two different scenarios of exception chaining in the segments.